lib/Solve.js

```diff
@@ -0,0 +1,66 @@
+// @flow
+
+const crypto = require('crypto');
+const path = require('path');
+const fs = require('fs-extra');
+const spawn = require('cross-spawn');
+
+function sha1(string) {
+  return crypto.createHash('sha1').update(string).digest('hex');
+}
+
+function getDependenciesCached(
+  generatedCodeDir /*: string */,
+  elmJsonPath /*: string */,
+  projectElmJson /*: any */
+) /*: { direct: { [string]: string }, indirect: { [string]: string } } */ {
+  const hash = sha1(
+    JSON.stringify({
+      dependencies: projectElmJson.dependencies,
+      'test-dependencies': projectElmJson['test-dependencies'],
+    })
+  );
+
+  const cacheFile = path.join(generatedCodeDir, `dependencies.${hash}.json`);
+
+  try {
+    return fs.readJsonSync(cacheFile);
+  } catch (_error) {
+    // Cache file does not exist or is malformed. Move on.
```

Comment:
One last question. What are you thoughts on checking whether `error.code === 'ENOENT'` and otherwise rethrowing the error? I guess it boils down to: is it better to sometimes crash when we could actually recover or to (sometime in the hypothetical future) start failing to cache elm.json dependencies and not notice?

Reply:
Very good question.

- Current approach: elm-test “self-heals” from a bad file, but on the other hand we might miss legit errors that we would have wanted to know about and code around.

- Your approach: elm-test fails hard on a bad file. Hopefully the user reports it to use. User needs to remove the file (or the whole `elm-stuff`) to get going again. We could mention all of this in the error message, though. Now that I think of it – that sounds like the best solution.

Reply:
Or we print a warning to stderr if there is a non `error.code === 'ENOENT'` exception but do not crash.

Reply:
I’m printing a warning now. Sample:

```
Ignoring bad dependencies cache file:

/Users/lydell/forks/node-test-runner/example-package/elm-stuff/generated-code/elm-community/elm-test/0.19.1-revision4/dependencies.d48585830c2c9ad9efb7f8be7173b55e9509bafd.json: Unexpected end of JSON input

Please report this issue: https://github.com/rtfeldman/node-test-runner/issues/new

elm-test 0.19.1-revision4
-------------------------

Running 16 tests. To reproduce these results, run: elm-test --fuzz 100 --seed 227195913342066

↓ TestsFailing
↓ basic expectations
…
```